sycl/plugins/cuda/CMakeLists.txt

```diff
@@ -21,6 +21,10 @@ add_library(pi_cuda SHARED
   "pi_cuda.cpp"
 )
 
+add_dependencies(pi_cuda
+  ocl-headers
+)
```

Comment:
Another solution could be to make the llvm/sycl/CMakeLists.txt  `OpenCL-Headers` library (used below) dependent on the custom target `ocl-headers` - though without testing I am not 100% sure if that would resolve this issue, too.

Reply:
It might not.
Basically we need to make sure that ocl-headers target to "built" before we start compilation of CUDA plug-in.

Reply:
To chime in, a bit late...

This was indeed the correct solution in my mind. Setting dependencies on interface libraries (which `OpenCL-Headers` is) is not possible. CMake just silently doesn't do anything for the dependency in that case. (I've been bit by this before...)